base/compiler/optimize.jl

```diff
@@ -548,40 +586,12 @@ function visit_conditional_successors(callback, lazypostdomtree::LazyPostDomtree
     return false
 end
 
-struct AugmentedDomtree
-    cfg::CFG
-    domtree::DomTree
-end
-
-mutable struct LazyAugmentedDomtree
-    const ir::IRCode
-    agdomtree::AugmentedDomtree
-    LazyAugmentedDomtree(ir::IRCode) = new(ir)
-end
-
-function get!(lazyagdomtree::LazyAugmentedDomtree)
-    isdefined(lazyagdomtree, :agdomtree) && return lazyagdomtree.agdomtree
-    ir = lazyagdomtree.ir
-    cfg = copy(ir.cfg)
-    # Add a virtual basic block to represent the exit
-    push!(cfg.blocks, BasicBlock(StmtRange(0:-1)))
-    for bb = 1:(length(cfg.blocks)-1)
-        terminator = ir[SSAValue(last(cfg.blocks[bb].stmts))][:stmt]
-        if isa(terminator, ReturnNode) && isdefined(terminator, :val)
```

Comment:
@Keno, do you think it's appropriate to always add a virtual edge, regardless of whether the actual return node is reachable? I believe this adjustment is necessary for `visit_conditional_successors` but I'm uncertain about its implications for the usage in `iterated_dominance_frontier` at L853.